pex/pyenv.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os
+import re
+import subprocess
+
+from pex.common import is_exe
+from pex.tracer import TRACER
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import attr  # vendor:skip
+    from typing import Iterator, List, Optional, Tuple
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class Pyenv(object):
+    root = attr.ib()  # type: str
+
+    @classmethod
+    def find(cls):
+        # type: () -> Optional[Pyenv]
+        """Finds the active pyenv installation if any."""
+        with TRACER.timed("Searching for pyenv root...", V=3):
+            pyenv_root = os.environ.get("PYENV_ROOT", "")
+            if not pyenv_root:
+                for path_entry in os.environ.get("PATH", "").split(os.pathsep):
+                    pyenv_exe = os.path.join(path_entry, "pyenv")
+                    if is_exe(pyenv_exe):
+                        process = subprocess.Popen(args=[pyenv_exe, "root"], stdout=subprocess.PIPE)
+                        stdout, _ = process.communicate()
+                        if process.returncode == 0:
+                            pyenv_root = str(stdout).strip()
+                            break
+
+            if pyenv_root:
+                pyenv = cls(pyenv_root)
+                TRACER.log("A pyenv installation was found: {}".format(pyenv), V=6)
+                return pyenv
+
+            TRACER.log("No pyenv installation was found.", V=6)
+            return None
+
+    @attr.s(frozen=True)
+    class Shim(object):
+        pyenv = attr.ib()  # type: Pyenv
+        path = attr.ib()  # type: str
+        name = attr.ib()  # type: str
+        major = attr.ib()  # type: Optional[str]
+        minor = attr.ib()  # type: Optional[str]
+
+        _SHIM_REGEX = re.compile(
+            r"""
+            ^
+            (?P<name>
+                python |
+                pypy
```

Comment:
What about things like jython? do we care?

Reply:
Pex only supports CPython and PyPY: 
https://github.com/pantsbuild/pex/blob/30bc1ab39ccec76934f8842bd333508d6e69ae43/pyproject.toml#L26-L29
https://github.com/pantsbuild/pex/blob/30bc1ab39ccec76934f8842bd333508d6e69ae43/pex/interpreter.py#L72-L75
https://github.com/pantsbuild/pex/blob/30bc1ab39ccec76934f8842bd333508d6e69ae43/pex/interpreter.py#L332-L342